packages/next/src/client/components/router-reducer/fetch-server-response.ts

```diff
@@ -34,8 +34,18 @@ export async function fetchServerResponse(
     headers[NEXT_ROUTER_PREFETCH] = '1'
   }
 
+  const isNextExport = process.env.__NEXT_CONFIG_OUTPUT === 'export'
+
   try {
-    const res = await fetch(url.toString(), {
+    let cloneUrl = new URL(url)
+    if (isNextExport) {
+      if (cloneUrl.pathname.endsWith('/')) {
+        cloneUrl.pathname += 'index.txt'
+      } else {
+        cloneUrl.pathname += '.txt'
+      }
+    }
+    const res = await fetch(cloneUrl, {
```

Comment:
```suggestion
    let finalUrl = url;
    if (isNextExport) {
      finalUrl = new URL(url);
      finalUrl.pathname += finalUrl.pathname.endsWith('/'))
        ? 'index.txt'
        : '.txt'
    }
    const res = await fetch(finalUrl, {
```

Now on non-export deploy there will be no extra `new URL`.

Reply:
Nice catch! Fixed in [d699b99](https://github.com/vercel/next.js/pull/47022/commits/d699b991c3b8bba04484a287f49a71595819fd43)